vulcanai2/models/basenetwork.py

```diff
@@ -327,47 +321,49 @@ def fit(self, train_loader, val_loader, epochs,
             if plot is True:
                 fig_number = plt.gcf().number + 1 if plt.fignum_exists(1) else 1
                 plt.show()
-                
-            for epoch in trange(self.epoch, epochs, desc='Epoch: ', ncols=80):
 
-                train_loss, train_acc = self._train_epoch()
-                
+            for epoch in trange(0, epochs, desc='Epoch: ', ncols=80):
+
+                train_loss, train_acc = self._train_epoch(train_loader, retain_graph)
+
                 valid_loss = valid_acc = np.nan
                 if epoch % valid_interv == 0:
-                    valid_loss, valid_acc = self._validate()
+                    valid_loss, valid_acc = self._validate(val_loader)
 
                 tqdm.write("\n Epoch {}:\n"
                            "Train Loss: {:.6f} | Test Loss: {:.6f} |"
                            "Train Acc: {:.4f} | Test Acc: {:.4f}".format(
-                                epoch,
-                                train_loss,
-                                valid_loss,
-                                train_acc,
-                                valid_acc
-                                ))
-
-                self.record['epoch'].append(epoch)
-                self.record['train_error'].append(train_loss)
-                self.record['train_accuracy'].append(train_acc)
-                self.record['validation_error'].append(valid_loss)
-                self.record['validation_accuracy'].append(valid_acc)
+                    self.epoch,
+                    train_loss,
+                    valid_loss,
+                    train_acc,
+                    valid_acc
+                ))
+
+                record['epoch'].append(self.epoch)
+                record['train_error'].append(train_loss)
+                record['train_accuracy'].append(train_acc)
+                record['validation_error'].append(valid_loss)
+                record['validation_accuracy'].append(valid_acc)
 
                 if plot is True:
                     plt.ion()
                     plt.figure(fig_number)
-                    display_record(record=self.record)
+                    # display_record(record=record)
```

Comment:
Why was this commented out?

Reply:
oops. I couldn't get it to work on my end (import errors) so I just gave up and commented it  out for testing purposes. Will uncomment and investigate.